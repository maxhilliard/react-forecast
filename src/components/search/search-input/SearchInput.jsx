import React from 'react';
import PropTypes from 'prop-types';

import styles from './SearchInput.css';

const SearchInput = ({ value, placeholder, handleInputChange }) => (
    <input
        placeholder={placeholder}
        type="text"
        name="search"
        value={value}
        onChange={handleInputChange}
        className={styles.searchInput}
    />
);

SearchInput.propTypes = {
    value: PropTypes.string,
    placeholder: PropTypes.string,
    handleInputChange: PropTypes.func.isRequired,
};

SearchInput.defaultProps = {
    value: '',
    placeholder: 'Enter location',
};

export default SearchInput;
