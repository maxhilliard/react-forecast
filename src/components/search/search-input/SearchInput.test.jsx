import React from 'react';
import { shallow } from 'enzyme';
import SearchInput from './SearchInput';

describe('SearchInput', () => {
    describe('snapshot', () => {
        it('should match snapshot', () => {
            const wrapper = shallow(<SearchInput handleInputChange={() => {}} />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('props', () => {
        it('should invoke handleInputChange prop on keypress', () => {
            const handleInputChangeSpy = jest.fn();
            const wrapper = shallow(<SearchInput handleInputChange={handleInputChangeSpy} />);

            const mockKeypressEvent = {
                target: { value: 'the-value' },
            };

            wrapper.find('input').simulate('change', mockKeypressEvent);

            expect(handleInputChangeSpy).toHaveBeenLastCalledWith(mockKeypressEvent);
        });
    });
});
