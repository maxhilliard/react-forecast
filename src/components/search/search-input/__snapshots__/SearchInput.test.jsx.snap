// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput snapshot should match snapshot 1`] = `
<input
  className="searchInput"
  name="search"
  onChange={[Function]}
  placeholder="Enter location"
  type="text"
  value=""
/>
`;
